VirtoCommerce.Storefront/Controllers/Api/ApiOrderController.cs

```diff
@@ -190,13 +190,15 @@ private async Task<CustomerOrder> GetOrderByNumber(string number)
         private async Task<orderModel.CustomerOrder> GetOrderDtoByNumber(string number)
         {
             var order = await _orderApi.GetByNumberAsync(number);
-            var authorizationResult = await _authorizationService.AuthorizeAsync(User, order, CanAccessOrderAuthorizationRequirement.PolicyName);
-
-            if (!authorizationResult.Succeeded)
+            var viewAllOrdersAuthorizationResult = await _authorizationService.AuthorizeAsync(User, null, SecurityConstants.Permissions.CanViewOrders);
+            if (!viewAllOrdersAuthorizationResult.Succeeded)
             {
-                throw new StorefrontException($"Order with number {{ number }} not found (or not belongs to current user)");
+                var authorizationResult = await _authorizationService.AuthorizeAsync(User, order, CanAccessOrderAuthorizationRequirement.PolicyName);
```

Comment:
Why you need this double authorization check? Why it can't be done  by one policy `CanAccessOrderAuthorizationRequirement`?

Reply:
Ok. I will think about this.